dd-java-agent/agent-iast/src/main/java/com/datadog/iast/sink/NoSameSiteCookieModuleImpl.java

```diff
@@ -0,0 +1,41 @@
+package com.datadog.iast.sink;
+
+import com.datadog.iast.model.Evidence;
+import com.datadog.iast.model.Location;
+import com.datadog.iast.model.Vulnerability;
+import com.datadog.iast.model.VulnerabilityType;
+import com.datadog.iast.overhead.Operations;
+import com.datadog.iast.util.CookieSecurityInfo;
+import datadog.trace.api.iast.sink.NoSameSiteCookieModule;
+import datadog.trace.bootstrap.instrumentation.api.AgentSpan;
+import datadog.trace.bootstrap.instrumentation.api.AgentTracer;
+
+public class NoSameSiteCookieModuleImpl extends SinkModuleBase
+    implements NoSameSiteCookieModule, ForCookieSecurityInfo {
+
+  @Override
+  public void onCookie(CookieSecurityInfo cookieSecurityInfo) {
```

Comment:
As I've commented before this could be:

```suggestion
  public void onCookie(CookieSecurityInfo cookieSecurityInfo) {
  if (!cookieSecurityInfo.isSameSiteStrict()) {
  checkCookieVulnerability(VulnerabilityType.NO_SAMESITE_COOKIE)
  }
  }
```